src/contention_point.cpp

```diff
@@ -2284,50 +2304,55 @@ bool ContentionPoint::eval_spec(Comp comp, IndexedExample* eg, time_range_t time
     for (unsigned int t = time_range.first; t <= time_range.second; t++) {
         metric_val lhs_m_val, rhs_m_val;
 
-        eval_lhs(comp.get_lhs(), eg, t, lhs_m_val);
+        eval_m_expr(comp.get_lhs(), eg, t, lhs_m_val);
         if (!lhs_m_val.valid) return false;
 
         eval_rhs(comp.get_rhs(), eg, t, rhs_m_val);
         if (!rhs_m_val.valid) return false;
 
         unsigned int lhs_val = lhs_m_val.value;
         unsigned int rhs_val = rhs_m_val.value;
-        bool res = eval_op(lhs_val, comp.get_op(), rhs_val);
+        bool res = Op::eval(lhs_val, comp.get_op(), rhs_val);
         if (!res) return false;
     }
 
     return true;
 }
 
-void ContentionPoint::eval_rhs(rhs_t rhs,
+void ContentionPoint::eval_rhs(Rhs* rhs,
                                IndexedExample* eg,
                                unsigned int time,
                                metric_val& res) const {
-    if (holds_alternative<m_expr_t>(rhs)) {
-        eval_m_expr(get<m_expr_t>(rhs), eg, time, res);
-    } else if (holds_alternative<Time>(rhs)) {
-        eval_Time(get<Time>(rhs), eg, time, res);
-    } else if (holds_alternative<unsigned int>(rhs)) {
+    MExpr* m_expr = dynamic_cast<MExpr*>(rhs);
+    Time* time_expr = dynamic_cast<Time*>(rhs);
+    Constant* const_expr = dynamic_cast<Constant*>(rhs);
+    if (m_expr) {
+        eval_m_expr(m_expr, eg, time, res);
+    } else if (time_expr) {
+        eval_Time(*time_expr, eg, time, res);
+    } else if (const_expr) {
         res.valid = true;
-        res.value = get<unsigned int>(rhs);
+        res.value = const_expr->get_coeff();
     } else {
         cout << "ContentionPoint::eval_rhs: invalid variant." << endl;
     }
```

Comment:
Do it the same way you have done for `get_expr(MExpr* lhs` (without else).

Reply:
[Remove else statement from ContentionPoint::eval_rhs](https://github.com/all-things-networking/fperf/pull/32/commits/bc9ae12f80cdc0bc38873d29497a861f15078406)